src/main/java/com/linecorp/armeria/server/metrics/ServiceMetricConsumer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.metrics;
+
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.linecorp.armeria.common.SessionProtocol;
+
+@FunctionalInterface
+public interface ServiceMetricConsumer {
+    Logger logger = LoggerFactory.getLogger(ServiceMetricConsumer.class);
+
+    /**
+     * Invoked for each request that has been processed
+     *
+     * @param sessionProtocol session protocol, currently only http-based session protocols are supported.
```

Comment:
`the {@link SessionProtocol} which the invocation has been performed on` ? (/cc @anuraaga)

I guess we don't need to state it's HTTP-only here. Perhaps we should say so in the `MetricCollectingService`?